src/components/Gif.js

```diff
@@ -0,0 +1,101 @@
+import React, {
+    Component
+} from "react";
+import {
+    addFav,
+    withdrawFav
+} from "../Containers/FavPage/action"
+import {
+    connect
+} from 'react-redux'
+
+import {Wraper, ImgStyled} from "./Gif.style";
+
+class Gifs extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            buttonText: "Fav",
+            fav: false
+        }
+        this.onClick = this.onClick.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.gif.length > 0) {
+            this.props.gif.map((image) => {
+                if (image.id === this.props.id) {
+                    this.setState({
+                        fav: true,
+                        buttonText: "NotFav"
+                    });
+                }
+            });
+        }
+    }
+
+    onClick() {
+        if (this.props.gif.length > 0) {
```

Comment:
You can avoid this `state` management leveraging the `fav` logic to the store.

Reply:
I will work on that.